atomix/storage/src/main/java/io/atomix/storage/journal/MappedJournalSegmentWriter.java

```diff
@@ -259,16 +262,20 @@ public void truncate(final long index) {
 
   @Override
   public void flush() {
-    mappedBuffer.force();
+    buffer.force();
   }
 
   @Override
   public void close() {
-    flush();
-    try {
-      BufferCleaner.freeBuffer(mappedBuffer);
-    } catch (final IOException e) {
-      throw new StorageException(e);
+    if (isOpen) {
+      isOpen = false;
+      flush();
+      BufferUtil.free(buffer);
```

Comment:
Nit: you can also use `org.agrona.IoUtil.unmap(buffer)`. It just calls `BufferUtil`, however imo it's a bit more readable in its intention.

Reply:
Make sense 👍  Still learning the agrona api. Thanks for sharing these